core/src/main/java/com/cognifide/aem/stubs/core/script/ConfigurableStubScriptManager.java

```diff
@@ -145,29 +169,50 @@ public String getRootPath() {
 
   @Override
   public String getExtension() {
-    return config.extension();
+    return config.scriptExtension();
   }
 
   @Override
   public void onChange(List<ResourceChange> changes) {
-    final List<String> scriptPaths = changes.stream()
-      .map(ResourceChange::getPath)
-      .filter(this::isRunnable)
-      .collect(Collectors.toList());
+    final List<String> mappingPaths = getMappingPaths(changes);
+
+    if (!mappingPaths.isEmpty()) {
+      resetAll(mappingPaths);
+      return;
```

Comment:
Why distinguishing paths which in the end are reseting all...

If reloading single mapping file is not possible maybe we should remove on change modes at all

Reply:
hmm. for mappings for sure it is needed - restart all